consensus/src/experimental/commit_phase.rs

```diff
@@ -0,0 +1,333 @@
+                                error = err,
+                                "Error occurred in signing commit vote."
+                            )
+                        }).unwrap();
+
+                    self.commit_msg_sender.send(
+                        CommitPhaseChannelMsgWrapper::CommitVote(self.author, new_ledger_info.clone(), sig)
+                        ).await.map_err(|err| {
+                            counters::ERROR_COUNT.inc();
+                            error!(
+                                epoch = new_ledger_info.commit_info().epoch(),
+                                error = err.to_string(),
+                                "Error occurred in sending the commit vote message."
+                            )
+                        }).ok(); // note that this message will also reach the node itself
+
+                    self.local_queue.push_back(CacheItem::new(
+                        vecblock,
+                        ledger_info,
+                    ));
+                }
+                msg = self.commit_msg_receiver.select_next_some() => {
+                    match msg {
+                        ConsensusMsg::CommitVoteMsg(request) => {
+                            monitor!(
+                                "process_commit_vote",
+                                self.process_commit_vote(&*request).await.map_err(|err| {
+                                    counters::ERROR_COUNT.inc();
+                                    error!(
+                                        error = err.to_string(),
+                                        "Error in processing commit vote."
+                                    )
+                                }).ok()
+                            );
+                        }
+                        ConsensusMsg::CommitDecisionMsg(request) => {
+                            monitor!(
+                                "process_commit_decision",
+                                self.process_commit_decision(*request).await.map_err(|err| {
+                                    counters::ERROR_COUNT.inc();
+                                    error!(
+                                        error = err.to_string(),
+                                        "Error in processing commit decision."
+                                    )
+                                }).ok()
+                            );
+                        }
+                        _ => {}
+                    };
+                }
+            }
+            let mut_local_queue = &mut self.local_queue;
+            let mut_local_cache = &mut self.local_cache;
+            loop {
+                let front = mut_local_queue.front().unwrap();
+                let front_chash = front
+                    .ledger_info_sig
+                    .ledger_info()
+                    .commit_info()
+                    .executed_state_id();
+                match mut_local_cache.entry(front_chash) {
```

Comment:
if let Some(li) = local_cache.remove()

Reply:
When the entry is not a valid QC yet, we do not remove the entry.